crates/matrix-sdk-base/src/client.rs

```diff
@@ -558,6 +564,52 @@ impl BaseClient {
         changes.account_data = account_data;
     }
 
+    /// User has left a room.
+    ///
+    /// Update the internal and cached state accordingly. Return the final Room.
+    pub async fn room_left(&self, room_id: &RoomId) -> Result<Room> {
+        let room = self.store.get_or_create_room(room_id, RoomType::Left).await;
+        if room.room_type() == RoomType::Left {
+            return Ok(room);
+        }
+
+        // needs to be updated first
```

Comment:
What needs to be updated first?

Reply:
I am curious, too. I assume it was meant that the room type should be updated first, but I don't get why?